crates/turborepo-lib/src/client.rs

```diff
@@ -12,14 +12,58 @@ use crate::{get_version, retry::retry_future};
 pub trait UserClient {
     fn set_token(&mut self, token: String);
     async fn get_user(&self) -> Result<UserResponse>;
+    async fn get_teams(&self) -> Result<TeamsResponse>;
```

Comment:
I noticed we're getting rid of the separate interfaces for link and login that are present in the Go impl. I'm fine as long as testability isn't negatively affected.

Reply:
Yeah it doesn't really make sense to have traits with overlapping methods, so we might as well combine them.